user_interface/EbikeUIController/functions.cpp

```diff
@@ -73,31 +73,43 @@ void pinModesInit() {
 //   Timer1.attachInterrupt(timerISR); // Attach timerISR function
 // }
 
-bool handleThrottleInput(int inputRequest, int &throttle, bool &activatedRegen) {
+bool handleThrottleInput(UserInputRequest inputRequest, int &throttle, bool &activatedRegen, RegenMethod regenMethod) {
   bool throttle_changed = false;
-  if (inputRequest == 1) {
-    // Increase throttle if not at maximum
-    if (throttle == 100) {
-      return false;
-    } else {
-      throttle += 5; // Increase throttle by 20
-      throttle_changed = 1; // Set throttle flag
-    }
-  } else if (inputRequest == 2) {
-    // Decrease throttle if not at minimum
-    if (throttle == 0) {
-      return false;
-    } else {
-      throttle -= 5; // Decrease throttle by 20
-      throttle_changed = 1; // Set throttle flag
-    }
+  switch (UserInputRequest) {
+    case INCREASE:
+      // Increase throttle if not at maximum
+      if (throttle == 100) {
+        return false;
+      } else {
+        throttle += 5; // Increase throttle by 20
+        throttle_changed = 1; // Set throttle flag
```

Comment:
I changed the type of this to bool, so these should be true/false. Technically arduino language is c++ instead of c, so its best practice to use bools instead of ints for this kind of thing